JASP-Engine/JASP/R/commonMachineLearningRegression.R

```diff
@@ -26,13 +26,21 @@
     testSetIndicator                  <- options[["testSetIndicatorVariable"]]
   variables.to.read         <- c(target, predictors, testSetIndicator)
   if (is.null(dataset)){
-    dataset <- .readDataSetToEnd(columns = variables.to.read, exclude.na.listwise = variables.to.read)
+    dataset <- .readAndAddCompleteRowIndices(dataset, variables.to.read)
   }
   if(length(unlist(options[["predictors"]])) > 0 && options[["target"]] != "" && options[["scaleEqualSD"]])
-    dataset[,.v(c(options[["predictors"]], options[["target"]]))] <- .scaleNumericData(dataset[,.v(c(options[["predictors"]], options[["target"]]))])
+    dataset[,.v(c(options[["predictors"]], options[["target"]]))] <- .scaleNumericData(dataset[,.v(c(options[["predictors"]], options[["target"]])), drop = FALSE])
   return(dataset)
 }
 
+.readAndAddCompleteRowIndices <- function(dataset, variables.to.read){
```

Comment:
This function looks deliciously compact. Maybe we could use this in Descriptives as well? 

Or maybe move this to common to give everybody easy access to this functionality?
@TimKDJ what do you think?

Reply:
Yeah we can definitely add it. In that case it would be better to take the `.readDataSetToEnd()` out of the function though, as this differs per analysis.

@koenderks why does `.readAndAddCompleteRowIndices()` need the `dataset` argument?